test/unit/controllers/todo-item.controller.test.ts

```diff
@@ -0,0 +1,260 @@
+      const expected = [
+        payloadWithId,
+        { title: 'do another thing', todoId: 2 }
+      ];
+      findTodoItems.resolves(expected);
+      findTodo.withArgs(todo.id as number).resolves(todo);
+      expect(await controller.getTodoItems(todo.id as number)).to.eql(expected);
+    });
+    it('returns an empty list if there are none', async () => {
+      const controller = new TodoItemController(todoItemRepo, todoRepo);
+      const findTodoItems = todoItemRepo.find as sinon.SinonStub;
+      const findTodo = todoRepo.findById as sinon.SinonStub;
+      const expected: TodoItem[] = [];
+      findTodoItems.resolves(expected);
+      findTodo.withArgs(todo.id as number).resolves(todo);
+      expect(await controller.getTodoItems(todo.id as number)).to.eql(expected);
+    });
+  });
+  describe('replaceTodoItem', () => {
+    it('replaces a todo item if it exists', async () => {
+      const controller = new TodoItemController(todoItemRepo, todoRepo);
+      const replace = todoItemRepo.deleteById as sinon.SinonStub;
+      const findTodoItem = todoItemRepo.findById as sinon.SinonStub;
+      const item = Object.assign({}, payload, { id: 1, todoId: 1 });
+      const newItem = Object.assign({}, payload, { title: 'title two' });
+      replace.withArgs(item.id as number, newItem).resolves(newItem);
+      findTodoItem.withArgs(item.id as number).resolves(todo);
+      expect(
+        await controller.replaceTodoItem(
+          item.id as number,
+          newItem,
+          todo.id as number
+        )
+      ).to.eql(newItem);
+    });
+
+    it('throws if it does not exist', async () => {
+      const controller = new TodoItemController(todoItemRepo, todoRepo);
+      const replace = todoItemRepo.deleteById as sinon.SinonStub;
+      const findTodoItem = todoItemRepo.findById as sinon.SinonStub;
+      const item = Object.assign({}, payload, { id: 1, todoId: 1 });
+      const newItem = Object.assign({}, payload, { title: 'title two' });
+      replace.withArgs(item.id as number, newItem).rejects();
+      findTodoItem.withArgs(item.id as number).resolves(null);
+      try {
+        await controller.replaceTodoItem(
+          item.id as number,
+          newItem,
+          todo.id as number
+        );
+      } catch (err) {
+        expect(err).to.match(/No todo item was found with id/);
+      }
+    });
+  });
+  describe('updateTodoItem', () => {
+    it('successfully updates existing items', async () => {
+      const controller = new TodoItemController(todoItemRepo, todoRepo);
+      const update = todoItemRepo.updateById as sinon.SinonStub;
+      const findTodoItem = todoItemRepo.findById as sinon.SinonStub;
+      const item = Object.assign({}, payload, { id: 1, todoId: 1 });
+      const newItem = Object.assign({}, payload, { title: 'title two' });
+      update.withArgs(item.id as number, newItem).resolves(newItem);
+      findTodoItem.withArgs(item.id as number).resolves(todo);
```

Comment:
I find these unit tests too similar to each other, the whole file looks like a big wall of text to me.

Could you please extract parts that are repeating into a shared test setup?

For example, I think that all `item`, `newItem`, `controller`, `update`, `findTodoItem`, `create`, etc. can be defined in a single `beforeEach` hook. 

It's ok that the hook configures more stuff than is used by each test, because creating extra stubs are easy. What's more important is to make the tests easy to read and understand, make it easy for the reader to figure out what is important in each test, what's the difference from other tests.